feat: Remove bookmark button from certain spots

AmarTrebinjac opened the pull request:
## Changes

<!--
### Describe what this PR does

- Short and concise, bullet points can help
- Screenshots if applicable can also help
-->

## Events

Did you introduce any new tracking events?

<!--
If yes please remove the comment HTML comment tags and fill the table below

Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

-->

## Experiment

Did you introduce any new experiments?

<!--
If yes please remove the comment HTML comment tags and follow the instructions below

Don't forget to send a message to the [#experiments](https://dailydotdev.slack.com/archives/C02JAUF8HJL/p1715175315620999) channel, following the template in slack, and adding a link to the message here.

> [!IMPORTANT]
> Please do not merge the PR until the experiment enrolment is approved.

-->

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

<!--
If relevant, please remove the comment HTML comment tags and fill the checkboxes below

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

-->

<!--
Copy and paste the below line outside the HTML comment tags to link this PR to the ticket in Jira

AS-{number} #done
or
MI-{number} #done
-->


### Preview domain
https://mi-662.preview.app.daily.dev

AmarTrebinjac commented on packages/shared/src/components/widgets/SimilarPosts.tsx:
```diff
@@ -52,58 +54,66 @@ const DefaultListItem = ({
+          />
+        )}
+      </div>
+      {!isPlus && (
```
This and adding the "isPlus" variable at the top is the only change. A lot of green due to changing return from () to {}

AmarTrebinjac commented on packages/shared/src/components/widgets/SimilarPosts.tsx:
```diff
@@ -52,58 +53,66 @@ const DefaultListItem = ({
+          />
+        )}
+      </div>
+      {!showPlusSubscription && (
```
adding the hook at the top and this is the only real change. A lot of diff due to changing component from implicit to explicit return.

capJavert reviewed (approved):
Looks good, just wondering if this should be merged before other bookmark stuff.

AmarTrebinjac commented:
> Looks good, just wondering if this should be merged before other bookmark stuff.

Nah, we created a intermediate branch. Just forgot to set it. Good catch!

rebelchris reviewed (commented):
Why did you need all the tab/feednav changes, not sure I follow those?
We only removing the bookmark button here no?

AmarTrebinjac commented:
> Why did you need all the tab/feednav changes, not sure I follow those? We only removing the bookmark button here no?

They're just cause I switched target branch from main to the intermediate. Updated the intermediate branch now

sshanzel commented on packages/shared/src/components/widgets/SimilarPosts.tsx:
```diff
@@ -52,58 +53,66 @@ const DefaultListItem = ({
-          className="text-text-tertiary"
-        />
+}: PostProps): ReactElement => {
+  const { showPlusSubscription } = usePlusSubscription();
```
In light of the discussion, it seems we don't have to check for Plus now 😅

AmarTrebinjac replied:
@sshanzel Done, and also removed tests, variables and functions that are no longer used.

ilasw commented:
@AmarTrebinjac we should move the target branch into main at this point, there is no reason to put into the intermediate one ✔️

rebelchris reviewed (approved):
Nice cleanup :)

sshanzel reviewed (approved):
Always love seeing code get deleted 🤩
